### PR TITLE
Allow multiple arguments for get/delete command

### PR DESCRIPTION
**Pull request by Joseph-Irving**

### CODE REVIEW

File: pkg/keto/cmd/get.go [hunk 1]
```diff
@@ -106,37 +105,36 @@ func getComputePoolCmdFunc(c *cobra.Command, args []string) error {
 		return err
 	}
 
-	name := ""
-	if len(args) == 1 {
-		name = args[0]
+	if len(args) == 0 {
+		args = []string{""}
```
Comment (vaijab) [inline pkg/keto/cmd/get.go]: I think you don't need to set this. `args` is already an empty slice of strings.

File: pkg/keto/cmd/get.go [hunk 2]
```diff
@@ -106,37 +105,36 @@ func getComputePoolCmdFunc(c *cobra.Command, args []string) error {
...
 		return err
 	}
 
-	return listComputePools(cli, clusterName, name)
+	return listComputePools(cli, clusterName, args)
 }
 
-func listMasterPools(cli *cli, clusterName, name string) error {
-	pools, err := cli.ctrl.GetMasterPools(clusterName, name)
+func listMasterPools(cli *cli, clusterName string, names []string) error {
```
Comment (vaijab) [inline pkg/keto/cmd/get.go]: Use variadic function args here for consistency.

File: pkg/keto/cmd/get.go [hunk 3]
```diff
@@ -106,37 +105,36 @@ func getComputePoolCmdFunc(c *cobra.Command, args []string) error {
...
+	pools, err := cli.ctrl.GetMasterPools(clusterName, names...)
 	if err != nil {
 		return err
 	}
 	return keto.PrintMasterPool(keto.GetPrinter(os.Stdout), pools, true)
 }
 
-func listComputePools(cli *cli, clusterName, name string) error {
-	pools, err := cli.ctrl.GetComputePools(clusterName, name)
+func listComputePools(cli *cli, clusterName string, names []string) error {
```
Comment (vaijab) [inline pkg/keto/cmd/get.go]: Use variadic function args here for consistency.

File: pkg/keto/cmd/get.go [hunk 4]
```diff
@@ -106,37 +105,36 @@ func getComputePoolCmdFunc(c *cobra.Command, args []string) error {
...
+	pools, err := cli.ctrl.GetComputePools(clusterName, names...)
 	if err != nil {
 		return err
 	}
 	return keto.PrintComputePool(keto.GetPrinter(os.Stdout), pools, true)
 }
 
-func listClusters(cli *cli, name string) error {
-	clusters, err := cli.ctrl.GetClusters(name)
+func listClusters(cli *cli, names []string) error {
```
Comment (vaijab) [inline pkg/keto/cmd/get.go]: Use variadic function args here for consistency.

File: pkg/keto/cmd/get.go [hunk 5]
```diff
@@ -44,17 +44,17 @@ var getClusterCmd = &cobra.Command{
 }
 
 func getClusterCmdFunc(c *cobra.Command, args []string) error {
-	name := ""
-	if len(args) == 1 {
-		name = args[0]
+
+	if len(args) == 0 {
+		args = []string{""}
```
Comment (vaijab) [inline pkg/keto/cmd/get.go]: I think you don't need to set this. args is already an empty slice of strings.

File: pkg/keto/cmd/get.go [hunk 6]
```diff
@@ -75,17 +75,16 @@ func getMasterPoolCmdFunc(c *cobra.Command, args []string) error {
 		return err
 	}
 
-	name := ""
-	if len(args) == 1 {
-		name = args[0]
+	if len(args) == 0 {
+		args = []string{""}
```
Comment (vaijab) [inline pkg/keto/cmd/get.go]: I think you don't need to set this. args is already an empty slice of strings.

File: pkg/controller/controller.go [hunk 7]
```diff
@@ -336,3 +379,39 @@ func (c *Controller) DeleteComputePool(clusterName, name string) error {
 	c.Logger.Printf("deleting computepool %q of cluster %q", name, clusterName)
 	return pooler.DeleteComputePool(clusterName, name)
 }
+
+func filterMasterPools(pool []*model.MasterPool, names []string) []*model.MasterPool {
+	pools := []*model.MasterPool{}
```
Comment (vaijab) [inline pkg/controller/controller.go]: Could do without allocating `pools`, but it's a minor thing.

File: pkg/controller/controller.go [hunk 8]
```diff
@@ -327,12 +338,83 @@ func (c *Controller) DeleteMasterPool(clusterName string) error {
...
+		}
+	}
+	return nil
+}
+
+func filterMasterPools(pools []*model.MasterPool, names []string) []*model.MasterPool {
+	pool := []*model.MasterPool{}
+
+	for _, p := range pools {
+
```
Comment (marcinc) [inline pkg/controller/controller.go]: Formatting - empty lines around if... same below.

File: pkg/controller/controller.go [hunk 9]
```diff
@@ -327,12 +338,76 @@ func (c *Controller) DeleteMasterPool(clusterName string) error {
...
+	}
+
+	if len(cluster) != 0 {
+		return cluster
+	}
+
+	return clusters
+}
+
+func matchNames(Name string, names []string) bool {
```
Comment (vaijab) [inline pkg/controller/controller.go]: Lowercase `Name`.

Comment (vaijab) [inline pkg/controller/controller.go]: Maybe call it something like `stringInSlice`, more generic.

File: pkg/keto/cmd/delete.go [hunk 10]
```diff
@@ -42,9 +42,14 @@ var deleteClusterCmd = &cobra.Command{
 }
 
 func deleteClusterCmdFunc(c *cobra.Command, args []string) error {
-	if len(args) != 1 {
+	if len(args) == 0 {
 		return errors.New("cluster name is not specified")
 	}
+
+	if len(args) > 1 {
+		return errors.New("You cannot delete more than one cluster at at time")
```
Comment (vaijab) [inline pkg/keto/cmd/delete.go]: Error string should start with lowercase char by convention.

**Comment by vaijab** [comment]
What about deleting multiple master pools? Why not allow multiple cluster deletion?

File: pkg/controller/controller.go [hunk 11]
```diff
@@ -327,12 +338,76 @@ func (c *Controller) DeleteMasterPool(clusterName string) error {
...
+	}
+
+	if len(pool) != 0 {
+		return pool
+	}
+
+	return pools
+}
+
+func filterClusters(clusters []*model.Cluster, names []string) []*model.Cluster {
```
Comment (marcinc) [inline pkg/controller/controller.go]: Same goes for this function.

File: pkg/controller/controller.go [hunk 12]
```diff
@@ -327,12 +338,76 @@ func (c *Controller) DeleteMasterPool(clusterName string) error {
...
+	cluster := []*model.Cluster{}
```
Comment (marcinc) [inline pkg/controller/controller.go]: this should probably be plural? eg. `filteredClusters`

File: pkg/controller/controller.go [hunk 13]
```diff
@@ -327,12 +338,76 @@ func (c *Controller) DeleteMasterPool(clusterName string) error {
...
+		err := pooler.DeleteComputePool(clusterName, name)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func filterMasterPools(pools []*model.MasterPool, names []string) []*model.MasterPool {
+	pool := []*model.MasterPool{}
```
Comment (marcinc) [inline pkg/controller/controller.go]: `filteredPools` maybe?

File: pkg/controller/controller.go [hunk 14]
```diff
@@ -265,43 +265,54 @@ func (c *Controller) computePoolExists(clusterName, name string, pooler cloudpro
...
-func (c *Controller) GetClusters(name string) ([]*model.Cluster, error) {
+func (c *Controller) GetClusters(names ...string) ([]*model.Cluster, error) {
 	cl, impl := c.Cloud.Clusters()
 	if !impl {
 		return []*model.Cluster{}, ErrNotImplemented
 	}
 	c.Logger.Printf("getting clusters")
-	return cl.GetClusters(name)
+
+	cluster, err := cl.GetClusters("")
```
Comment (marcinc) [inline pkg/controller/controller.go]: `clusters` probably better?

File: pkg/controller/controller.go (hunk 13 above)
Comment (Joseph-Irving) [inline pkg/controller/controller.go]: filtered is an adjective while filter is a verb. I'd normally prefer a verb for a function

File: pkg/controller/controller.go [hunk 15]
```diff
@@ -327,12 +338,76 @@ func (c *Controller) DeleteMasterPool(clusterName string) error {
...
+	}
+
+	if len(pool) != 0 {
+		return pool
+	}
+	return pools
+
+}
+
+func filterComputePools(pools []*model.ComputePool, names []string) []*model.ComputePool {
```
Comment (Joseph-Irving) [inline pkg/controller/controller.go]: I did look into doing this but it ended up being requiring more lines of code, as you had to convert []*model.ComputePool into []*model.NodePool, filter it and then convert it back into []*model.ComputePool

File: pkg/controller/controller.go (hunk 13 above)
Comment (marcinc) [inline pkg/controller/controller.go]: I was referring to variable name :)

**Pull request merged**